google/cloud/spanner_v1/pool.py

```diff
@@ -328,7 +514,17 @@ def put(self, session):
         :param session: the session being returned.
         """
         try:
+            if self._borrowed_sessions.__contains__(session):
+                self._borrowed_sessions.remove(session)
             self._sessions.put_nowait(session)
+            self._traces.pop(session._session_id, None)
```

Comment:
Here we can not extract out the common method, because there is a very slight difference in few properties here. Like Burstpool has `self. target_size` while other two pool has `self.size` . Pinging pool does `self._sessions.put_nowait((_NOW() + self._delta, session))` while other two pool does `self._sessions.put_nowait(session)` .

Reply:
Okay got it. Thanks